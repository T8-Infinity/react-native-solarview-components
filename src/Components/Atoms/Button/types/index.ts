import { TouchableOpacityProps } from 'react-native';

export interface IPropsButton extends TouchableOpacityProps {
  title: string | undefined;
  loading?: boolean;
  colorTitle?: string;
  onPress: (() => void) | undefined;
  backgroundColor?: string;
  outline?: boolean;
  disabled?: boolean;
  borderColor?: string;
  testID?: string;
}
