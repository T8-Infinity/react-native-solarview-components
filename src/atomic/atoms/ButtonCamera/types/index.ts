import { TouchableOpacityProps } from 'react-native';

export interface IButtonCameraProps extends TouchableOpacityProps {
  testID?: string;
  icon: string;
  onPress: () => void;
  color: string;
}
