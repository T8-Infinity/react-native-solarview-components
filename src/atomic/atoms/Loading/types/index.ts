import { ActivityIndicatorProps } from 'react-native';

export interface Props extends ActivityIndicatorProps {
  testID?: string;
  size: 'large' | 'small';
  color?: string;
}
