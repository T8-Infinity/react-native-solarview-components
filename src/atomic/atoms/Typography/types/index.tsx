import { fonts } from '@theme/fonts';
import { size } from '@theme/size';

export interface ITextProps {
  text: string;
  fontSize: keyof typeof size;
  color?: string;
  variant: keyof typeof fonts;
}
