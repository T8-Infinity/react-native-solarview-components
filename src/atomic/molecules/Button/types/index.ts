import { TouchableOpacityProps } from 'react-native';

export interface IPropsButton extends TouchableOpacityProps {
  title: string;
  loading?: boolean;
  colorTitle?: string;
  onPress: () => void;
  backgroundColor?: string;
  outline?: boolean;
  disabled?: boolean;
  borderColor?: string;
  testID?: string;
  icon: string;
}
