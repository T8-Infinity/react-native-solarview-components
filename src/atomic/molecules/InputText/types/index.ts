import { TextInput, TextInputProps } from 'react-native';

export interface InputTextProps extends TextInputProps {
  title: string;
  value: string;
  label?: string;
  icon?: string;
  secret?: boolean;
  touched?: boolean;
  errorMessage?: string;
  onChangeText: (e: any) => void;
  handlePressIcon?: () => void;
  colorTitle?: string;
}

export type Props = InputTextProps & {
  inputRef?: React.RefObject<TextInput>;
};

export interface InputTypesProps {
  colorTitle?: string;
  touched?: boolean;
}
