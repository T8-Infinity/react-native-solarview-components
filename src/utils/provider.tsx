import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@theme/index';

export const provider: React.FC = ({ children }: any) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
