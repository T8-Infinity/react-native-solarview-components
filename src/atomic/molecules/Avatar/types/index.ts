import { IButtonCameraProps } from '@atomic/atoms/ButtonCamera/types';

export interface IAvatarProps {
  testID?: string;
  url: string;
  withLabel?: string;
  type: 'integrator' | 'owner';
  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  color: string;
  icon: string;
}

export type IProps = IAvatarProps & IButtonCameraProps;
