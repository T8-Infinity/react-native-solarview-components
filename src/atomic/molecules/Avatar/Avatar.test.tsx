import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { provider as wrapper } from '@utils/provider';
import { IProps } from './types';

import { Avatar } from '.';

describe('Molecules/Avatar', () => {
  const elAvatar = 'Avatar';

  const props: IProps = {
    color: '',
    icon: 'camera',
    onPress: jest.fn(),
    size: 'lg',
    type: 'integrator',
    url: '',
  };

  const setup = () => {
    const utils = render(<Avatar {...props} />, {
      wrapper,
    });
    const sut = utils.getByTestId(`${elAvatar}`);
    const sutImage = utils.getByTestId(`${elAvatar}_image`);
    const sutButton = utils.getByTestId(`${elAvatar}_button`);

    return {
      sut,
      sutImage,
      sutButton,
      ...utils,
    };
  };

  it(`should render component ${elAvatar}`, () => {
    const { sut } = setup();
    expect(sut).toBeTruthy();
  });

  it(`should press button ${elAvatar}`, () => {
    const { sutButton } = setup();
    fireEvent(sutButton, 'press');
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it(`should be able render image ${elAvatar}`, () => {
    const { sutImage } = setup();
    expect(sutImage).toBeTruthy();
  });

  it(`should be able to test conditional ${elAvatar}`, async () => {
    const { sutImage } = setup();
    props.icon.length > 0;
    expect(sutImage).toBeTruthy();
  });

  it('should render snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
