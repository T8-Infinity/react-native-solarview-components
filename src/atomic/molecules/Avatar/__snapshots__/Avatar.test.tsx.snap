// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Avatar should render snapshot 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
  testID="Avatar"
>
  <Image
    resizeMode="contain"
    size="lg"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#008868",
          "borderRadius": 128,
          "borderWidth": 3,
          "height": 192,
          "width": 192,
        },
      ]
    }
    testID="Avatar_image"
    type="integrator"
  />
  <View
    size="lg"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 48,
          "bottom": 14,
          "height": 48,
          "justifyContent": "center",
          "left": 140,
          "position": "absolute",
          "width": 48,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
        ]
      }
      testID="Avatar_button"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="Avatar_button_touch"
      >
        <View
          color=""
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#008868",
                "borderRadius": 48,
                "height": 48,
                "justifyContent": "center",
                "width": 48,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                },
                Array [
                  Object {},
                ],
                Object {
                  "fontFamily": "material-community",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ó°„€
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
